Philips.CodeAnalysis.MaintainabilityAnalyzers/AvoidTaskResultCodeFixProvider.cs

```diff
@@ -0,0 +1,71 @@
+﻿// © 2021 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Text;
+using Philips.CodeAnalysis.Common;
+
+namespace Philips.CodeAnalysis.MaintainabilityAnalyzers
+{
+
+	[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(AvoidTaskResultCodeFixProvider)), Shared]
+	public class AvoidTaskResultCodeFixProvider : CodeFixProvider
+	{
+		#region Non-Public Data Members
+
+		private const string Title = "Use await";
+
+		#endregion
+
+		#region Non-Public Properties/Methods
+
+		#endregion
+
+		#region Public Interface
+
+		public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(Helper.ToDiagnosticId(DiagnosticIds.AvoidTaskResult));
+		public sealed override FixAllProvider GetFixAllProvider()
+		{
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+			Diagnostic diagnostic = context.Diagnostics.First();
+			TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+			MemberAccessExpressionSyntax resultExpression = root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf().OfType<MemberAccessExpressionSyntax>().First();
+
+			if (resultExpression is null)
+			{
+				return;
+			}
+
+			context.RegisterCodeFix(
+				CodeAction.Create(
+					title: Title,
+					createChangedDocument: c => ReplaceWithAwait(context.Document, resultExpression, c),
+					equivalenceKey: Title),
+				diagnostic);
+		}
+
+		private async Task<Document> ReplaceWithAwait(Document document, MemberAccessExpressionSyntax resultExpression, CancellationToken cancellationToken)
+		{

```

Comment:
Should we worry about the case where the containing method isn't async?

```
void Foo(Task<T> task)
{
  var result = task.Result;
}
```

will not compile after the fix

Reply:
This was intentionally not supported.  (It will open a can of worms to cascade the async stuff.)  The goal was to point the author in the right direction.  The ensuing compiler error is clear about considering to make the method async, which will likely trigger additional code changes.